### PR TITLE
Improve camel wmq component to allow bytebuffer as body

### PR DESCRIPTION
**Pull request by bkuchcik**
For our internal development, we need to improve WMQConsumer :
Today it only allow to use String as body output but we need to add simple bytebuffer too.

### CODE REVIEW

File: components/camel-wmq/pom.xml
Comment (jbonofre) [inline components/camel-wmq/pom.xml]: Please remove this section as it's customer specific.

File: components/camel-wmq/src/main/java/org/apacheextras/camel/component/wmq/MQHeaderListCreationException.java
Comment (jbonofre) [inline components/camel-wmq/src/main/java/org/apacheextras/camel/component/wmq/MQHeaderListCreationException.java]: Why not using a `IllegalStateException` instead of a custom one ?

**Review by jbonofre — CHANGES_REQUESTED** [review]
Please cleanup. Thanks.

File: components/camel-wmq/src/main/java/org/apacheextras/camel/component/wmq/MQHeaderListCreationException.java
Comment (bkuchcik) [reply components/camel-wmq/src/main/java/org/apacheextras/camel/component/wmq/MQHeaderListCreationException.java]: When I started writing Junit test, I found a very complex error linked to MQHeaderList (when I create a new MQHeaderList instance). 

I think that this specific exception can help to understand the problem (if you forget a class in the classpath, currently not in the module dependency, you get a NoSucheMethodError).

I can add more information about that in the exception message to help developers to understand that problem.

Comment (bkuchcik) [reply components/camel-wmq/src/main/java/org/apacheextras/camel/component/wmq/MQHeaderListCreationException.java]: And i forgot the super call so I need to update it

Comment (bkuchcik) [reply components/camel-wmq/src/main/java/org/apacheextras/camel/component/wmq/MQHeaderListCreationException.java]: You are right, I wanted to handle checked exception and IllegalStateException can be enough.

I add another exception message for a case really hard to debug that I found when I wrote my unit cases (and it's the reason why I added the HeaderListFactory)

**Review by jbonofre — CHANGES_REQUESTED** [review]
Thanks for the changes ! Can you please squash the commits ?

File: components/camel-wmq/src/main/java/org/apacheextras/camel/component/wmq/MQHeaderListCreationException.java
Comment (jbonofre) [reply components/camel-wmq/src/main/java/org/apacheextras/camel/component/wmq/MQHeaderListCreationException.java]: +1

Comment (bkuchcik) [reply components/camel-wmq/src/main/java/org/apacheextras/camel/component/wmq/MQHeaderListCreationException.java]: can you wait before merging ? I want to do one more test before it

**Comment by jbonofre** [comment]
No worries, I won't merge before the squash anyway ;)

File: components/camel-wmq/src/main/java/org/apacheextras/camel/component/wmq/MQHeaderListCreationException.java
Comment (jbonofre) [reply components/camel-wmq/src/main/java/org/apacheextras/camel/component/wmq/MQHeaderListCreationException.java]: Sure, no problem, I won't merge before the commits squashing.

**Comment by jbonofre** [comment]
Let me know when squash is done, I will merge then. Thanks !

**Comment by bkuchcik** [comment]
Squash done !

File: components/camel-wmq/pom.xml
Comment (jbonofre) [inline components/camel-wmq/pom.xml:31]: By the way, I forgot in the previous review round. The version should be `2.19-SNAPSHOT` to be part of the reactor.

I will do the fix for you.

**Review by jbonofre — APPROVED** [review]

File: components/camel-wmq/pom.xml
Comment (jbonofre) [reply components/camel-wmq/pom.xml:31]: Oh you fixed on the branch :( I cherry-pick on master then.

**Pull request merged**
